### DISCUSSION TITLE
Can't login with password hash nor reset the pass via other means

### DISCUSSION THREAD

**Discussion by ubuntupunk** [kanboard/kanboard#5256] [General]
### Checklist

- [x ] I verified that Kanboard is correctly installed
- [x ] I verified that the problem does not come from a plugin
- [x ] I verified that the problem is not already reported
- [x ] I understand that Kanboard is in maintenance mode. It doesn't mean it's abandoned, but there is no significant feature development

### Actual behaviour
using hashed password fails
password reset link fails

### Expected behaviour

expected to be able to reset my admin password using the [method described in issue 853](https://github.com/kanboard/kanboard/issues/853)

expected to reset password via password reset
link fails, nothing happens

### Steps to reproduce
generate password hash
alter mysql table for user and add pass hash

or

request password reset.

### Logs
tail
```
[02-May-2023 13:23:35 Africa/Johannesburg] Kanboard: user kanadmin authentication failure with IP address: 165.0.131.172
[02-May-2023 13:25:21 Africa/Johannesburg] Kanboard: user kanadmin authentication failure with IP address: 165.0.131.172
```

### Configuration

- Kanboard version: 
- Database type and version: mariadb 15.1
- PHP version: 8.1.17
- OS: linux
- Browser: chromium

**Comment by cptsanifair** [comment]
Hi,

i tryed the solution from issue #853 on latest kanboard releast and interactive shell with PHP8.2.5 and 7.3.10 successful.

No Mail send:
- did u configure smtp for your Kanboard?
- did your account have a email-address?
- have it access to the smtp server?


can you try this short solution (from your kanboard-root folder) ?

```php
<?php

DEFINE('DS', DIRECTORY_SEPARATOR );
require_once __DIR__ . DS . 'data' . DS . 'config.php';

try {
    $conn = new PDO("mysql:" .
            "host=" . DB_HOSTNAME . "; " . 
            ( DB_PORT != null ?: "port= ". DB_PORT ."; ").
            "dbname=" . DB_NAME, DB_USERNAME, DB_PASSWORD);

    $conn->exec("SET CHARACTER SET utf8");      // Sets encoding UTF-8
    $conn->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);

    $sql = "UPDATE `" . DB_NAME . "`.`users` SET `users`.`password`= :newpassword WHERE `users`.`username`='admin'";

    $statement = $conn->prepare($sql);
    $statement->bindValue(":newpassword", password_hash('admin', PASSWORD_DEFAULT));

    echo $statement->execute();
    unset($conn);
}
catch(PDOException $e) {
  echo $e->getMessage();
}
```

regards
Cpt

**Comment by fguillot** [comment]
You can also use Kanboard command line tool to reset the password, for example: `./cli user:reset-password admin`

For reference: https://docs.kanboard.org/v1/admin/cli/#reset-user-password

**Comment by ubuntupunk** [comment]
Thanks for this, was worried the app lacks support.
